### PR TITLE
Merge master → xplat

### PR DESCRIPTION
**Pull request by rainersigwald**
Ignore until I work through the remaining build/test issues on all platforms, please.

### CODE REVIEW

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

File: src/Shared/FileUtilities.cs [hunk 1]
```diff
@@ -48,34 +65,32 @@ internal static partial class FileUtilities
...
         /// <summary>
-        /// Check if we are running unit tests (under some kind of test runner). If so, set the flag and come up with a
-        /// (potentially) fake executable path. Generally, the path will be used to find the config file, but also to
-        /// start msbuild.exe for remote nodes.
+        /// Check if we are running unit tests (under some kind of test runner) or in Visual Studio. If so, set the 
+        /// flag and come up with a (potentially) fake executable path. Generally, the path will be used to find 
+        /// the config file, but also to start msbuild.exe for remote nodes.
         /// </summary>
-        private static void GetTestExecutionInfo()
+        private static void GetExecutionInfo()
```
Comment (rainersigwald) [inline src/Shared/FileUtilities.cs]: Changes here caused some confusion, leading to #669.

**Comment by rainersigwald** [comment]
For as long as this took, I just scrolled through the whole diff and didn't see any real problem areas. Looking at my individual commits would point out most of the things I had to fix post-merge.

@AndyGerlicher @radical Can you take a look at the refactored import-search-path things? That caused merge conflicts since it got changed in xplat, ported to master, and then changed.

@AndyGerlicher @cdmihai I think this is ready for review. The OS X build passes on my machine but continually fails in CI doing nuget restore. I'll kick it again (also that will run Linux tests now, yay!).

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by cdmihai** [comment]
I disabled the linux tests until I figure out why xunit is not reporting their failures.

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test this please

**Comment by AndyGerlicher** [comment]
:shipit:

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build please

**Comment by rainersigwald** [comment]
I wish the restore wasn't quite so flaky, but . . . 🎉

**Pull request merged**